core/trino-main/src/main/java/io/trino/cost/StatisticRange.java

```diff
@@ -122,7 +123,17 @@ public double overlapPercentWith(StatisticRange other)
         if (isInfinite(length()) && isFinite(lengthOfIntersect)) {
             return INFINITE_TO_FINITE_RANGE_INTERSECT_OVERLAP_HEURISTIC_FACTOR;
         }
+
         if (lengthOfIntersect > 0) {
+            double thisDensity = this.distinctValues / length();
+            double otherDensity = other.distinctValues / other.length();
+            double minDensity = minExcludeNaN(thisDensity, otherDensity);
+
+            if (!isNaN(thisDensity) && !isNaN(otherDensity)
+                    && isFinite(length()) && isFinite(other.length())
```

Comment:
Why do we check that the lengths are finite ?
I think we want to skip `lengthOfIntersect == length()` case here